api/api/src/main/scala/uk/ac/wellcome/platform/api/services/FiltersAndAggregationsBuilder.scala

```diff
@@ -28,15 +41,17 @@ import scala.collection.immutable._
   * - `pairedFilters`: a list of all of the given filters which are paired to one
   *   of the given aggregations. These can be used as the ES post-query filters.
   */
-class FiltersAndAggregationsBuilder(
-  aggregationRequests: List[WorkAggregationRequest],
-  filters: List[WorkFilter],
-  requestToAggregation: WorkAggregationRequest => Aggregation,
-  filterToQuery: WorkFilter => Query) {
+trait FiltersAndAggregationsBuilder[Filter, AggregationRequest] {
```

Comment:
Pleasantly surprised by how nicely this came out re parametrising

Reply:
Yeah, it reflects the similarity in the way we've implemented Works and Images. ✨